lib/gyro/liquidgen/generator.rb

```diff
@@ -0,0 +1,120 @@
+# Copyright 2016 - Niji
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'gyro/xcdatamodel/parser'
+require 'gyro/liquidgen/whitespace_patch'
+require 'liquid'
+require 'pathname'
+
+module Gyro
+  module Liquidgen
+    # Generates arbitrary output from the input datamodel, using a Liquid template provided by the user
+    #
+    class Generator
+      # INCLUDES #############################################################
+
+      include Gyro::XCDataModel::Parser
+
+      # PUBLIC METHODS #######################################################
+
+      # Declare some custom Liquid Filters used by the template, then render it
+      module CustomFilters
```

Comment:
Might be worth moving that in a dedicated file (if you do, don't forget to add the `require` line at the top of this `generator.rb` to include it)